### PR TITLE
dfdlx:trace now goes to the logger under debug, added a test to show …

### PR DESCRIPTION
**Pull request by alexanderrevello**
…dfdlx:trace no longer goes to stderr

### CODE REVIEW

File: daffodil-runtime1/src/test/scala/org/apache/daffodil/dpath/TestDFDLXFunctions.scala
Comment (stevedlawrence) [inline daffodil-runtime1/src/test/scala/org/apache/daffodil/dpath/TestDFDLXFunctions.scala:71]: Tests that run the daffodil CLI are part of the `IntegrationTest` configuration, and must be in the `daffodil-cli/src/it/` directory and can be run with `sbt IntegrationTest/test`. The tests in that directory aso already have infrastructure for testing the CLI, so much of the logic you have (e.g. slashes, .bat, capturing output) is already implemented.  I'd recommend creating a new test in that repo using an existing CLI test as an example.

File: daffodil-runtime1/src/test/scala/org/apache/daffodil/dpath/TestDFDLXFunctions.scala
Comment (stevedlawrence) [inline daffodil-runtime1/src/test/scala/org/apache/daffodil/dpath/TestDFDLXFunctions.scala:66]: This doesn't test that we actually get trace output. We want to make sure that we get the right error message to stderror. Mentioned in another comment, we can do that with ExpectIt and the existing test infrastructure.

File: daffodil-runtime1/src/test/scala/org/apache/daffodil/dpath/TestDFDLXFunctions.scala
Comment (stevedlawrence) [inline daffodil-runtime1/src/test/scala/org/apache/daffodil/dpath/TestDFDLXFunctions.scala:61]: Your schema calls `dfdlx:trace` in the `dfdl:inputValueCalc` property. That property is only evaluated when parsing, but this is an unparse test, so it will never see the output. `dfdlx:trace` doesn't care about parse vs unparse, so it seems reasonable to just make this a parse test.

File: daffodil-runtime1/src/main/resources/test/validation/trace_output.dfdl.xsd
Comment (stevedlawrence) [inline daffodil-runtime1/src/main/resources/test/validation/trace_output.dfdl.xsd:43]: We aren't that strict about this, but we usually prefer files end with a newline. There should be a setting in your IDE to make sure files are saved with a newline.

**Review by stevedlawrence — CHANGES_REQUESTED** [review]

**Comment by tuxji** [comment]
@alexanderrevello, how did these additional commits (written by other contributors) appear in your branch?  We no longer can see only your own changes (we see other contributors' changes mixed with your changes) so we cannot review your pull request anymore. 

Please don't deviate from Apache Daffodil's [code contributor workflow](https://cwiki.apache.org/confluence/display/DAFFODIL/Code+Contributor+Workflow).  You will have to close this pull request, make a new pull request with a single commit on a fresh branch, wait for reviewers to request changes, add new commits to address these requests, and repeat until you have two +1's approving your pull request.  Then you can rebase your branch upon the latest changes in the main branch, squash your commits down to a single commit, and wait for a Daffodil committer to merge your pull request.

**Pull request closed without merging**
